src/versionbits.h

```diff
@@ -79,11 +79,11 @@ struct VersionBitsCache
     void Clear();
 };
 
-/** Get the BIP9 state for a given deployment at the current tip. */
+/** Get the BIP9 state for a given deployment for the block after pindexPrev. */
 ThresholdState VersionBitsState(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::DeploymentPos pos, VersionBitsCache& cache);
-/** Get the numerical statistics for the BIP9 state for a given deployment at the current tip. */
+/** Get the numerical statistics for the BIP9 state for a given deployment as at pindexPrev. */
```

Comment:
In "versionbits: correct doxygen comments"

Nit: perhaps it's worth stopping using `pindexPrev` here if it's giving statistics for the specified block rather than its successor?

Reply:
It's giving the statistics for the retarget period that includes `pindexPrev`'s successor, up to and including `pindexPrev` if `pindexPrev` is in the same retarget period, but not if `pindexPrev` was the last block in the previous retarget period. (That means you can't actually query the statistics for a fully completed retarget period -- you'll always get statistics for between 0 and 2015 blocks worth of signalling, never the full 2016 blocks. Or whatever the period is)

I've been confused about this until recently, so not sure how to describe it well

Reply:
"Get statistics for the BIP9 state for the measuring window that inlcudes pindexPrev's successor." ?

Feel free to disregard this.